src/renderer/components/+config-limit-ranges/limit-range-details.tsx

```diff
@@ -0,0 +1,134 @@
+import "./limit-range-details.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { KubeObjectDetailsProps } from "../kube-object";
+import { ILimitRangeItem, LimitRange, Resource } from "../../api/endpoints/limit-range.api";
+import { kubeObjectDetailRegistry } from "../../api/kube-object-detail-registry";
+import { KubeObjectMeta } from "../kube-object/kube-object-meta";
+import { DrawerItem } from "../drawer/drawer-item";
+import { Badge } from "../badge";
+
+interface Props extends KubeObjectDetailsProps<LimitRange> {
+}
+
+function renderLimits(limit: ILimitRangeItem, resource: string): JSX.Element {
+
+  return (
+    <div>
+      {limit.defaultRequest &&
+      Object.keys(limit.defaultRequest).filter(key => key === resource).map(resourceName => (
+        <Badge key={resourceName} label={`defaultRequest: ${limit.defaultRequest[resourceName]}`}/>
+      ))
+      }
```

Comment:
It would be (imo) easier to read this if these were split out. It is also strange to see a filter select only one item and then map over it. Furthermore there is high amount of overlap between each of these limits. So factoring out the common items would be even better.

Something like the following:

```tsx
function renderLimit(limit: LimitRangeItem, part: LimitPart, resource: Resource) {
  const resourceLimit = limit[part]?.[resource];

  if (!resourceLimit) {
    return null;
  }

  return <Badge key={resource} label={`${part}: ${resourceLimit}`}/>;
}
```

And then this return could very easily be:
```
return (
  <>
    {renderLimit(limit, LimitPart.DEFAULT_REQUEST, resource)}
    {renderLimit(limit, LimitPart.DEFAULT, resource)}
    {renderLimit(limit, LimitPart.MAX_LIMIT_RATIO, resource)}
    {renderLimit(limit, LimitPart.MAX, resource)}
    {renderLimit(limit, LimitPart.MIN, resource)}
  </>
);
```

Reply:
Hi @Nokel81, thank you for your review. Now, the code looks more readable and cleaner.